### PR TITLE
If it rains, you put on a coat. Rack. (Coatrack buffs.)

### PR DESCRIPTION
**Pull request by alsoandanswer**
Basically, adds support for every concievable coat for the coat rack. (and ponchos.)

That means no more hard-sprited coats. They simply recycle the on-mob sprite.

dreamseeker_R7ftEoZcNR

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Review by Chada-1 — APPROVED** [review]
💟

**Review by Casper3667 — APPROVED** [review]

File: code/game/objects/structures/coatrack.dm [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	icon_state = "coatrack"
+	var/obj/item/clothing/coat
+	var/obj/item/clothing/head/hat
+	var/list/custom_sprites = list(/obj/item/clothing/head/beret/security, /obj/item/clothing/accessory/poncho/tajarancloak) // Custom manual sprite override.
+
+/obj/structure/coatrack/attack_hand(mob/user as mob)
+	if(!ishuman(user))
+		return
+	if(user.incapacitated())
+		return
```
Comment (Arrow768) [inline code/game/objects/structures/coatrack.dm:14]: You should use the use_check / use_check_and_message proc instead possibly with optional flags.

File: code/game/objects/structures/coatrack.dm [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		user.put_in_hands(coat)
+		coat = null
+		update_icon()
+	else if(hat)
+		user.visible_message("[user] takes [hat] off \the [src].", SPAN_NOTICE("You take [hat] off the \the [src]."))
+		user.put_in_hands(hat)
+		hat = null
+		update_icon()
+
+/obj/structure/coatrack/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (Arrow768) [inline code/game/objects/structures/coatrack.dm:28]: You should use a use_check / use_check_and_message proc here possibly with optional flags.

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**